vectorizer.py

```diff
@@ -24,17 +24,19 @@ class Vectorizer:
     model: AutoModel
     tokenizer: AutoTokenizer
     cuda: bool
+    mps: bool
     cuda_core: str
     model_type: str
 
-    def __init__(self, model_path: str, cuda_support: bool, cuda_core: str, model_type: str, architecture: str):
+    def __init__(self, model_path: str, cuda_support: bool, cuda_core: str, model_type: str, architecture: str, ismps: bool):
```

Comment:
Lets stick to `snake_case`, this concatenated variable names can be confusing, `mps_support` might be a good variable name and is consistent with `cuda_support`.